wire-schema/src/main/java/com/squareup/wire/schema/internal/parser/OptionElement.kt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2013 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire.schema.internal.parser
+
+import com.squareup.wire.schema.internal.Util
+import com.squareup.wire.schema.internal.Util.appendIndented
+
+data class OptionElement(
+  val name: String,
+  val kind: Kind,
+  val value: Any,
+  private val isParenthesized: Boolean
+) {
+  enum class Kind {
+    STRING,
+    BOOLEAN,
+    NUMBER,
+    ENUM,
+    MAP,
+    LIST,
+    OPTION
+  }
+
+  fun toSchema(): String = buildString {
+    when (kind) {
+      Kind.STRING -> return """${formatName()} = "$value""""
+      Kind.BOOLEAN,
+      Kind.NUMBER,
+      Kind.ENUM -> return "${formatName()} = $value"
+      Kind.OPTION -> {
+        // Treat nested options as non-parenthesized always, prevents double parentheses.
+        val optionValue = (value as OptionElement).run {
+          OptionElement.create(name, kind, value)
+        }
+        append("${formatName()}.${optionValue.toSchema()}")
+      }
+      Kind.MAP -> {
+        append("${formatName()} = {\n")
+        formatOptionMap(this, value as Map<String, *>)
+        append('}')
+      }
+      Kind.LIST -> {
+        append("${formatName()} = ")
+        Util.appendOptions(this, value as List<OptionElement>)
+      }
+    }
+  }
+
+  fun toSchemaDeclaration() = "option ${toSchema()};\n"
+
+  private fun formatName() = if (isParenthesized) "($name)" else name
+
+  companion object {
+    @JvmOverloads
+    fun create(
+      name: String,
+      kind: Kind,
+      value: Any,
+      isParenthesized: Boolean = false
+    ) = OptionElement(name, kind, value, isParenthesized)
+
+    private fun formatOptionMap(
+      builder: StringBuilder,
+      valueMap: Map<String, *>
+    ) {
+      valueMap.entries.forEach { entry ->
+        val endl = if (entry != valueMap.entries.last()) "," else ""
```

Comment:
switch to `forEachIndexed` and use a prefix comma for `i > 0`

Reply:
Maybe you can't because of `appendIndented` actually. I'd feel better if this was index based somehow.

Reply:
it can be done!